In [24]:
print('Takumi House & Kailen Santos')

Takumi House & Kailen Santos


In [163]:
from matplotlib import pyplot as plt
import matplotlib as mp
import numpy as np
import pandas as pd
from sklearn.cluster import KMeans
import sklearn.manifold
from scipy import cluster
import seaborn as sns
import sklearn.datasets as sk_data
import sklearn.metrics as metrics
import requests
import json
import nltk.classify.util
from nltk.classify import NaiveBayesClassifier
from nltk.corpus import stopwords
from nltk.stem.porter import PorterStemmer
from nltk import word_tokenize
import re
import json
import pickle
%matplotlib inline

In [164]:
##load in pickled dataframe of all the reviews

#try:
#    with open('df_reviews','rb') as f:
#        df_reviews = pickle.load(f)
        
#except FileNotFoundError:
#    with open('df_reviews','wb') as f:
#        pickle.dump(df_reviews, f)

In [165]:
##load in pickled stemmed data for the review text
try:
    with open('stemmed_data','rb') as f:
        stemmed_data = pickle.load(f)
        
except FileNotFoundError:
    stemmed_data = [" ".join(SnowballStemmer("english", ignore_stopwords=True).stem(word)  
        for sent in sent_tokenize(message)
        for word in word_tokenize(sent))
        for message in df_reviews['text']]
    
    with open('stemmed_data','wb') as f:
        pickle.dump(stemmed_data, f)

In [226]:
df = pd.read_csv('Restaurant_Inspections.csv',header=0)
# df = df[df.City == 'Las Vegas']
# df = (df[df['Inspection Result'] == 'Closed with Fees']).append(df[df['Inspection Result'] == 'Closed without Fees'])

df.shape

#with open ('Resaurant_Inspections.csv') as file:
#   for line in file:
#      if (line[6]=='Las Vegas' && line[]

/Users/takumihouse/anaconda3/lib/python3.5/site-packages/IPython/core/interactiveshell.py:2717: DtypeWarning: Columns (18) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


(123919, 23)

In [227]:
df = df[df.City == 'Las Vegas']
df.shape

(94700, 23)

In [228]:
categories = df['Category Name'].unique().tolist()
print(categories)

['Pantry', 'Special Kitchen', 'Garde Manger', 'Kitchen Bakery', 'Bar / Tavern', 'Buffet', 'Snack Bar', 'Restaurant', 'Banquet Support', 'Portable Unit', 'Banquet Kitchen', 'Portable Bar', 'Meat/Poultry/Seafood', 'Vegetable Prep', 'Caterer', 'Barbeque', 'Confection', 'Concessions', 'Produce Market', 'Institutional Food Service', 'Elementary School Kitchen', 'Self-Service Food Truck', 'Bakery Sales', 'Food Trucks / Mobile Vendor', 'Grocery Store Sampling', 'Beer Bar', 'Childcare Kitchens', "Farmer's Market", 'Main Kitchen', 'Frozen Meat Sales']


In [229]:
categories.remove('Elementary School Kitchen')
categories.remove('Grocery Store Sampling')
categories.remove('Farmer\'s Market')
categories.remove('Caterer')
categories.remove('Concessions')
categories.remove('Institutional Food Service')
categories.remove('Banquet Support')
categories.remove('Snack Bar')

In [230]:
df = df[df['Category Name'].isin(categories)]
df.shape

(82656, 23)

In [231]:
coord_strings = df['Location 1'].tolist()

In [232]:
import ast
xcoords = []
ycoords = []
for coord in coord_strings:
    xcoords.append(ast.literal_eval(coord)[0])
    ycoords.append(ast.literal_eval(coord)[1])

In [233]:
count = 0
for idx, ycoord in enumerate(ycoords):
    if ycoord > 0:
        ycoords[idx] = ycoord * -1
        count += 1
print('{} coordinates fixed.'.format(count))

510 coordinates fixed.


In [234]:
df['Coordinates'] = list(zip(xcoords, ycoords))
df.head()

Serial Number Permit Number                 Restaurant Name  \
11     DA0001715     PR0000426  CAESARS AUGUSTUS BUTLER PANTRY   
12     DA0001716     PR0000426  CAESARS AUGUSTUS BUTLER PANTRY   
13     DA0001728     PR0000427  CAESARS DIAMOND LOUNGE KITCHEN   
14     DA0001773     PR0000434         CAESARS BAKERY WAREWASH   
15     DA0001776     PR0000434         CAESARS BAKERY WAREWASH   

                    Location Name    Category Name                Address  \
11  CAESARS PALACE HOTEL & CASINO           Pantry  3570 S Las Vegas Blvd   
12  CAESARS PALACE HOTEL & CASINO           Pantry  3570 S Las Vegas Blvd   
13  CAESARS PALACE HOTEL & CASINO  Special Kitchen  3570 S Las Vegas Blvd   
14  CAESARS PALACE HOTEL & CASINO  Special Kitchen  3570 S Las Vegas Blvd   
15  CAESARS PALACE HOTEL & CASINO  Special Kitchen  3570 S Las Vegas Blvd   

         City   State         Zip  Current Demerits  \
11  Las Vegas  Nevada  89109-8924                 3   
12  Las Vegas  Nevada  89109-8924                 3   
13  Las Vegas  Nevada  89109-8924                 8   
14  Las Vegas  Nevada  89109-8924                 3   
15  Las Vegas  Nevada  89109-8924                 3   

               ...             Employee ID     Inspection Type  \
11             ...               EE7000417  Routine Inspection   
12             ...               EE7000567  Routine Inspection   
13             ...               EE7000417  Routine Inspection   
14             ...               EE7000417  Routine Inspection   
15             ...               EE7000392  Routine Inspection   

   Inspection Demerits Inspection Grade Permit Status Inspection Result  \
11                   0                A           NaN         Compliant   
12                   2                A           NaN         Compliant   
13                   4                A           NaN         Compliant   
14                   7                A           NaN         Compliant   
15                   2                A           NaN         Compliant   

    Violations          Record Updated                  Location 1  \
11         NaN  02/21/2013 10:26:12 PM  (36.1161559, -115.1750576)   
12         NaN  02/21/2013 10:26:12 PM  (36.1161559, -115.1750576)   
13          14  02/21/2013 10:26:12 PM  (36.1161559, -115.1750576)   
14    14232831  01/22/2014 02:03:44 PM  (36.1161559, -115.1750576)   
15         NaN  01/22/2014 02:03:44 PM  (36.1161559, -115.1750576)   

                   Coordinates  
11  (36.1161559, -115.1750576)  
12  (36.1161559, -115.1750576)  
13  (36.1161559, -115.1750576)  
14  (36.1161559, -115.1750576)  
15  (36.1161559, -115.1750576)  

[5 rows x 24 columns]

In [236]:
coordinates_list = df['Coordinates'].tolist()
xcoords = []
ycoords = []
for coordinates in coordinates_list:
    xcoords.append(coordinates[0])
    ycoords.append(coordinates[1])
df['X'] = xcoords
df['Y'] = ycoords

In [237]:
df = df[df['Y'] < -114.5]
df = df[df['Y'] > -115.6]
df_clean = df.reset_index(drop = True)

In [238]:
coordinates_list = df_clean['Coordinates'].tolist()
xcoords = []
ycoords = []
for coordinates in coordinates_list:
    xcoords.append(coordinates[0])
    ycoords.append(coordinates[1])

In [239]:
df = df_clean[df_clean['Date Current']==df_clean['Inspection Date']]
# df['XY'] = list(zip(df.X, df.Y))
df.head()

Serial Number Permit Number                            Restaurant Name  \
60      DA0003413     PR0000492   CAESARS SEASONAL VENUS POOL BAR  DELETED   
187     DA0009343     PR0000728  MIRAGE 1 OAK SMALL BAR 2 ORIGINAL DELETED   
243     DA0011310     PR0000730     MIRAGE 1 OAK MULTILEVEL BAR 4  DELETED   
306     DA0017310     PR0000631                  Golden Gate Beverage Cart   
410     DA0023482     PR0001171                         Aramark Nacho Cart   

                     Location Name  Category Name                Address  \
60   CAESARS PALACE HOTEL & CASINO  Portable Unit  3570 S Las Vegas Blvd   
187          MIRAGE HOTEL & CASINO   Bar / Tavern  3400 S Las Vegas Blvd   
243          MIRAGE HOTEL & CASINO   Bar / Tavern  3400 S Las Vegas Blvd   
306     Golden Gate Hotel & Casino  Portable Unit           1 Fremont St   
410       ARAMARK @ CASHMAN CENTER  Portable Unit   850 N LAS VEGAS Blvd   

          City   State         Zip  Current Demerits     ...      \
60   Las Vegas  Nevada  89109-8924                 4     ...       
187  Las Vegas  Nevada  89109-8923                10     ...       
243  Las Vegas  Nevada  89109-8923                 3     ...       
306  Las Vegas  Nevada  89101-5601                 0     ...       
410  Las Vegas  Nevada       89101                 0     ...       

    Inspection Demerits Inspection Grade Permit Status Inspection Result  \
60                    4                A           NaN         Compliant   
187                  10                A           NaN         Compliant   
243                   3                A           NaN         Compliant   
306                   0                A           NaN         Compliant   
410                   0                A           NaN         Compliant   

    Violations          Record Updated                  Location 1  \
60         114  02/21/2013 10:26:12 PM  (36.1161559, -115.1750576)   
187      64114  02/21/2013 10:26:12 PM  (36.1206015, -115.1768382)   
243         13  02/21/2013 10:26:12 PM  (36.1206015, -115.1768382)   
306        NaN  02/21/2013 10:26:12 PM  (36.1713005, -115.1462312)   
410        NaN  02/21/2013 10:26:12 PM   (36.1800444, -115.131106)   

                    Coordinates          X           Y  
60   (36.1161559, -115.1750576)  36.116156 -115.175058  
187  (36.1206015, -115.1768382)  36.120601 -115.176838  
243  (36.1206015, -115.1768382)  36.120601 -115.176838  
306  (36.1713005, -115.1462312)  36.171301 -115.146231  
410   (36.1800444, -115.131106)  36.180044 -115.131106  

[5 rows x 26 columns]

In [240]:
# df_sorted = df.sort(columns = 'XY', axis = 0)
# df_sorted.head(n=40)

In [241]:
df_clean = df.reset_index(drop=True)
df_clean.head()

Serial Number Permit Number                            Restaurant Name  \
0     DA0003413     PR0000492   CAESARS SEASONAL VENUS POOL BAR  DELETED   
1     DA0009343     PR0000728  MIRAGE 1 OAK SMALL BAR 2 ORIGINAL DELETED   
2     DA0011310     PR0000730     MIRAGE 1 OAK MULTILEVEL BAR 4  DELETED   
3     DA0017310     PR0000631                  Golden Gate Beverage Cart   
4     DA0023482     PR0001171                         Aramark Nacho Cart   

                   Location Name  Category Name                Address  \
0  CAESARS PALACE HOTEL & CASINO  Portable Unit  3570 S Las Vegas Blvd   
1          MIRAGE HOTEL & CASINO   Bar / Tavern  3400 S Las Vegas Blvd   
2          MIRAGE HOTEL & CASINO   Bar / Tavern  3400 S Las Vegas Blvd   
3     Golden Gate Hotel & Casino  Portable Unit           1 Fremont St   
4       ARAMARK @ CASHMAN CENTER  Portable Unit   850 N LAS VEGAS Blvd   

        City   State         Zip  Current Demerits     ...      \
0  Las Vegas  Nevada  89109-8924                 4     ...       
1  Las Vegas  Nevada  89109-8923                10     ...       
2  Las Vegas  Nevada  89109-8923                 3     ...       
3  Las Vegas  Nevada  89101-5601                 0     ...       
4  Las Vegas  Nevada       89101                 0     ...       

  Inspection Demerits Inspection Grade Permit Status Inspection Result  \
0                   4                A           NaN         Compliant   
1                  10                A           NaN         Compliant   
2                   3                A           NaN         Compliant   
3                   0                A           NaN         Compliant   
4                   0                A           NaN         Compliant   

  Violations          Record Updated                  Location 1  \
0        114  02/21/2013 10:26:12 PM  (36.1161559, -115.1750576)   
1      64114  02/21/2013 10:26:12 PM  (36.1206015, -115.1768382)   
2         13  02/21/2013 10:26:12 PM  (36.1206015, -115.1768382)   
3        NaN  02/21/2013 10:26:12 PM  (36.1713005, -115.1462312)   
4        NaN  02/21/2013 10:26:12 PM   (36.1800444, -115.131106)   

                  Coordinates          X           Y  
0  (36.1161559, -115.1750576)  36.116156 -115.175058  
1  (36.1206015, -115.1768382)  36.120601 -115.176838  
2  (36.1206015, -115.1768382)  36.120601 -115.176838  
3  (36.1713005, -115.1462312)  36.171301 -115.146231  
4   (36.1800444, -115.131106)  36.180044 -115.131106  

[5 rows x 26 columns]

In [246]:
# df_agg = df_clean.groupby('Restaurant Name',as_index=False).agg(lambda x: x.tolist())
df_agg = df_clean.copy()

In [247]:
df_agg.head()

Serial Number Permit Number                            Restaurant Name  \
0     DA0003413     PR0000492   CAESARS SEASONAL VENUS POOL BAR  DELETED   
1     DA0009343     PR0000728  MIRAGE 1 OAK SMALL BAR 2 ORIGINAL DELETED   
2     DA0011310     PR0000730     MIRAGE 1 OAK MULTILEVEL BAR 4  DELETED   
3     DA0017310     PR0000631                  Golden Gate Beverage Cart   
4     DA0023482     PR0001171                         Aramark Nacho Cart   

                   Location Name  Category Name                Address  \
0  CAESARS PALACE HOTEL & CASINO  Portable Unit  3570 S Las Vegas Blvd   
1          MIRAGE HOTEL & CASINO   Bar / Tavern  3400 S Las Vegas Blvd   
2          MIRAGE HOTEL & CASINO   Bar / Tavern  3400 S Las Vegas Blvd   
3     Golden Gate Hotel & Casino  Portable Unit           1 Fremont St   
4       ARAMARK @ CASHMAN CENTER  Portable Unit   850 N LAS VEGAS Blvd   

        City   State         Zip  Current Demerits     ...      \
0  Las Vegas  Nevada  89109-8924                 4     ...       
1  Las Vegas  Nevada  89109-8923                10     ...       
2  Las Vegas  Nevada  89109-8923                 3     ...       
3  Las Vegas  Nevada  89101-5601                 0     ...       
4  Las Vegas  Nevada       89101                 0     ...       

  Inspection Demerits Inspection Grade Permit Status Inspection Result  \
0                   4                A           NaN         Compliant   
1                  10                A           NaN         Compliant   
2                   3                A           NaN         Compliant   
3                   0                A           NaN         Compliant   
4                   0                A           NaN         Compliant   

  Violations          Record Updated                  Location 1  \
0        114  02/21/2013 10:26:12 PM  (36.1161559, -115.1750576)   
1      64114  02/21/2013 10:26:12 PM  (36.1206015, -115.1768382)   
2         13  02/21/2013 10:26:12 PM  (36.1206015, -115.1768382)   
3        NaN  02/21/2013 10:26:12 PM  (36.1713005, -115.1462312)   
4        NaN  02/21/2013 10:26:12 PM   (36.1800444, -115.131106)   

                  Coordinates          X           Y  
0  (36.1161559, -115.1750576)  36.116156 -115.175058  
1  (36.1206015, -115.1768382)  36.120601 -115.176838  
2  (36.1206015, -115.1768382)  36.120601 -115.176838  
3  (36.1713005, -115.1462312)  36.171301 -115.146231  
4   (36.1800444, -115.131106)  36.180044 -115.131106  

[5 rows x 26 columns]

In [248]:
dfcategories = df_agg.columns
dfcategories

Index(['Serial Number', 'Permit Number', 'Restaurant Name', 'Location Name',
       'Category Name', 'Address', 'City', 'State', 'Zip', 'Current Demerits',
       'Current Grade', 'Date Current', 'Inspection Date', 'Inspection Time',
       'Employee ID', 'Inspection Type', 'Inspection Demerits',
       'Inspection Grade', 'Permit Status', 'Inspection Result', 'Violations',
       'Record Updated', 'Location 1', 'Coordinates', 'X', 'Y'],
      dtype='object')

In [249]:
# drop unnecessary columns for categorization step later
df_agg = df_agg.drop(['Serial Number'], axis = 1)
df_agg = df_agg.drop(['City'], axis = 1)
df_agg = df_agg.drop(['State'], axis = 1)
df_agg = df_agg.drop(['Zip'], axis = 1)
df_agg = df_agg.drop(['Inspection Time'], axis = 1)
df_agg = df_agg.drop(['Employee ID'], axis = 1)
df_agg = df_agg.drop(['Inspection Type'], axis = 1)
df_agg = df_agg.drop(['Location 1'], axis = 1)
df_agg = df_agg.drop(['Coordinates'], axis = 1)
df_agg = df_agg.drop(['Record Updated'], axis = 1)
df_agg = df_agg.drop(['Inspection Date'], axis = 1)

In [250]:
df_agg.head()

Permit Number                            Restaurant Name  \
0     PR0000492   CAESARS SEASONAL VENUS POOL BAR  DELETED   
1     PR0000728  MIRAGE 1 OAK SMALL BAR 2 ORIGINAL DELETED   
2     PR0000730     MIRAGE 1 OAK MULTILEVEL BAR 4  DELETED   
3     PR0000631                  Golden Gate Beverage Cart   
4     PR0001171                         Aramark Nacho Cart   

                   Location Name  Category Name                Address  \
0  CAESARS PALACE HOTEL & CASINO  Portable Unit  3570 S Las Vegas Blvd   
1          MIRAGE HOTEL & CASINO   Bar / Tavern  3400 S Las Vegas Blvd   
2          MIRAGE HOTEL & CASINO   Bar / Tavern  3400 S Las Vegas Blvd   
3     Golden Gate Hotel & Casino  Portable Unit           1 Fremont St   
4       ARAMARK @ CASHMAN CENTER  Portable Unit   850 N LAS VEGAS Blvd   

   Current Demerits Current Grade            Date Current  \
0                 4             A  05/17/2010 12:00:00 AM   
1                10             A  07/30/2010 12:00:00 AM   
2                 3             A  07/30/2010 12:00:00 AM   
3                 0             A  03/16/2010 12:00:00 AM   
4                 0             A  05/04/2011 12:00:00 AM   

   Inspection Demerits Inspection Grade Permit Status Inspection Result  \
0                    4                A           NaN         Compliant   
1                   10                A           NaN         Compliant   
2                    3                A           NaN         Compliant   
3                    0                A           NaN         Compliant   
4                    0                A           NaN         Compliant   

  Violations          X           Y  
0        114  36.116156 -115.175058  
1      64114  36.120601 -115.176838  
2         13  36.120601 -115.176838  
3        NaN  36.171301 -115.146231  
4        NaN  36.180044 -115.131106

In [251]:
df_agg.shape

(12926, 15)

In [53]:
dfcategories = df_agg.columns
for x in dfcategories[1:]:
    for y in range(len(df_agg[x])):
        df_agg[x][y] = list(set(df_agg[x][y]))
        df_agg[x][y] = df_agg[x][y][0]

TypeError: 'float' object is not iterable

In [252]:
df_agg.head()

Permit Number                            Restaurant Name  \
0     PR0000492   CAESARS SEASONAL VENUS POOL BAR  DELETED   
1     PR0000728  MIRAGE 1 OAK SMALL BAR 2 ORIGINAL DELETED   
2     PR0000730     MIRAGE 1 OAK MULTILEVEL BAR 4  DELETED   
3     PR0000631                  Golden Gate Beverage Cart   
4     PR0001171                         Aramark Nacho Cart   

                   Location Name  Category Name                Address  \
0  CAESARS PALACE HOTEL & CASINO  Portable Unit  3570 S Las Vegas Blvd   
1          MIRAGE HOTEL & CASINO   Bar / Tavern  3400 S Las Vegas Blvd   
2          MIRAGE HOTEL & CASINO   Bar / Tavern  3400 S Las Vegas Blvd   
3     Golden Gate Hotel & Casino  Portable Unit           1 Fremont St   
4       ARAMARK @ CASHMAN CENTER  Portable Unit   850 N LAS VEGAS Blvd   

   Current Demerits Current Grade            Date Current  \
0                 4             A  05/17/2010 12:00:00 AM   
1                10             A  07/30/2010 12:00:00 AM   
2                 3             A  07/30/2010 12:00:00 AM   
3                 0             A  03/16/2010 12:00:00 AM   
4                 0             A  05/04/2011 12:00:00 AM   

   Inspection Demerits Inspection Grade Permit Status Inspection Result  \
0                    4                A           NaN         Compliant   
1                   10                A           NaN         Compliant   
2                    3                A           NaN         Compliant   
3                    0                A           NaN         Compliant   
4                    0                A           NaN         Compliant   

  Violations          X           Y  
0        114  36.116156 -115.175058  
1      64114  36.120601 -115.176838  
2         13  36.120601 -115.176838  
3        NaN  36.171301 -115.146231  
4        NaN  36.180044 -115.131106

In [253]:
try:
    with open('df_agg_revs','rb') as f:
        df_agg_revs = pickle.load(f)
        
except FileNotFoundError:
    with open('df_agg_revs','wb') as f:
        pickle.dump(non_0_sick_scores, f)

In [254]:
sick_ids = []
for x in df_agg_revs['business_id'].items():
    sick_ids.append(x[1])
print(sick_ids)

['F8ezW6hYgmQYBmXOVjhytQ', 'atLmZg76uDg0oKhKjHLMyg', 'y7mody_dVe_2K8-wnnPtUw', 'Su2-lf8TbowpHZcS3XIRfQ', 'xoYY0F5hrs00h285tWmZZA', 'KuXGis6uUuhpSKmD1s5Zvw', 'y_L32uJc_sPupdjhb2YO-g', 'stZBSJLsrwrikYUkZ4aAAA', 'lSS9k3Fbs8E8qEVTy4-d5w', 'RnuCCKK-5Ezfz0xv0F2kag', 'dVWP9QbfrKkMgSySiaY4HA', 'hsgTsHqun5t2vZyGNMILZg', '91ohFfAbLvc7HcI68OHy3w', '-CIZh4JOZuOEbSzLVrAIGw', 'i7HvZ1yTTIaR6bfa_3jGEw', 'SsTxjxo8qvqBMvan1rzNzg', '7fuiTVuFt7CTAQRpFQf8DA', 'fLQ9XFlHtYHdAKCS4cSMag', 'XbnnsZKL8AohGRzo1dCFCQ', 'mdOsYpHnL_faguImkNbheA', 'ivi_LQZswT5xBpqXsuqK1Q', 'gcY4e5P1aac-Cj0NePNmPQ', 'XDk9FglRpw4TzhWejKwP4g', 'wLmMGlGCRdNSxArKca6DJA', 'aTwYIgWQUfoa2HobcjodTw', 'AFDFOs2GaIyUArGyEUDQzQ', 'botAdj1CUl2wA_Dz1Y-kIw', 'c8b7niizd3jigDlq-z_E6Q', 'VRpis8ZOVDFyO0WZAW5Rog', 'ylYhDUedhhG4WlcjvFq4Cw', 'PBkdlqw53cc2AZsfFD8AEQ', '94HshaJUV3DLvXkvbCbZqg', 'tFU2Js_nbIZOrnKfYJYBBg', 'KAxevTSEIuw99MUKnlzw5Q', 'RsI0wYUbWIHrHtOGZ54stg', 'xZoyDn7ekD4ue_liS_nqMA', 'RUz4d1D83xRRXcc-ffkM5g', 'ySpNcdz0tSpaPNqefAbfOw', 'pcIPHOoQZu

In [255]:
file = 'yelp_academic_dataset_business.json'
business_locations = {}
addresses = []
names=[]
with open(file) as f:
    for line in f:
        r_line = json.loads(line)
        if r_line['business_id'] in set(sick_ids):
            business_locations.update({r_line['business_id']:(r_line['full_address']).rsplit('\n', 5)[0].lower()})
            addresses.append(r_line['full_address'].rsplit('\n', 5)[0].lower())
            names.append(r_line['name'].lower())
addresses = np.asarray(addresses)
df_agg_revs['address']=addresses
df_agg_revs['name']=names
#print(business_locations)

In [257]:
df_agg_revs.head()

business_id                                              stars  \
0  F8ezW6hYgmQYBmXOVjhytQ      [3, 4, 2, 4, 3, 4, 1, 1, 1, 3, 4, 5, 1, 3, 4]   
1  atLmZg76uDg0oKhKjHLMyg  [2, 5, 1, 1, 5, 4, 3, 1, 1, 2, 1, 5, 3, 2, 2, ...   
2  y7mody_dVe_2K8-wnnPtUw  [5, 4, 4, 4, 5, 5, 5, 2, 4, 5, 5, 4, 5, 4, 4, ...   
3  Su2-lf8TbowpHZcS3XIRfQ                                       [1, 4, 3, 4]   
4  xoYY0F5hrs00h285tWmZZA  [4, 4, 4, 4, 3, 4, 4, 3, 4, 5, 2, 3, 3, 2, 4, ...   

                                                text  \
0  [This Jack's is not on the glitzy Las Vegas St...   
1  [Open 24 hours a day. Carna asada burrito is p...   
2  [The first time I was here, I ordered the Hawa...   
3  [Long unneCessary waits sometimes!\n ..... Poo...   
4  [New pizza place at the Wynn, named appropriat...   

                                               votes  \
0  [{'funny': 3, 'useful': 4, 'cool': 4}, {'funny...   
1  [{'funny': 0, 'useful': 1, 'cool': 0}, {'funny...   
2  [{'funny': 0, 'useful': 2, 'cool': 2}, {'funny...   
3  [{'funny': 1, 'useful': 2, 'cool': 0}, {'funny...   
4  [{'funny': 1, 'useful': 2, 'cool': 1}, {'funny...   

                      address                   name  
0     1916 village center cir           chianti cafe  
1       2737 las vegas blvd n             rigos taco  
2             2312 e craig rd  mulligans bar & grill  
3       3100 e lake mead blvd         domino's pizza  
4  1824 e lake mead boulevard  carl's jr restaurants

In [258]:
df_lv_reviews = df_agg_revs.copy()
df_lv_reviews = df_lv_reviews[['address', 'name']]
df_lv_reviews.head()

address                   name
0     1916 village center cir           chianti cafe
1       2737 las vegas blvd n             rigos taco
2             2312 e craig rd  mulligans bar & grill
3       3100 e lake mead blvd         domino's pizza
4  1824 e lake mead boulevard  carl's jr restaurants

In [259]:
df_lv_reviews.shape

(4910, 2)

In [284]:
df_reviews_search = df_lv_reviews.set_index('address')
df_reviews_search.head()

name
address                                          
1916 village center cir              chianti cafe
2737 las vegas blvd n                  rigos taco
2312 e craig rd             mulligans bar & grill
3100 e lake mead blvd              domino's pizza
1824 e lake mead boulevard  carl's jr restaurants

In [285]:
df_reviews_search.shape

(4910, 1)

In [271]:
df_lv_violations = df_agg.copy()
df_lv_violations = df_lv_violations[['Restaurant Name', 'Address']]
df_lv_violations['Restaurant Name'] = df_lv_violations['Restaurant Name'].str.lower()
df_lv_violations['Address'] = df_lv_violations['Address'].str.lower()
df_lv_violations.head()

Restaurant Name                Address
0   caesars seasonal venus pool bar  deleted  3570 s las vegas blvd
1  mirage 1 oak small bar 2 original deleted  3400 s las vegas blvd
2     mirage 1 oak multilevel bar 4  deleted  3400 s las vegas blvd
3                  golden gate beverage cart           1 fremont st
4                         aramark nacho cart   850 n las vegas blvd

In [272]:
df_lv_violations.shape

(12926, 2)

In [286]:
df_violations_search = df_lv_violations.set_index('Address')
df_violations_search.head()
# len(df_lv_violations['Address'].unique())

Restaurant Name
Address                                                         
3570 s las vegas blvd   caesars seasonal venus pool bar  deleted
3400 s las vegas blvd  mirage 1 oak small bar 2 original deleted
3400 s las vegas blvd     mirage 1 oak multilevel bar 4  deleted
1 fremont st                           golden gate beverage cart
850 n las vegas blvd                          aramark nacho cart

In [287]:
df_violations_search.shape

(12926, 1)

In [265]:
# bad_add = []
# for add_list in df_lv_violations['Address'].tolist():
#     if (len(set(add_list)) > 1):
#         bad_add.append(add_list)      

In [312]:
# def mergeDFs():
df_merged = pd.merge(df_reviews_search, df_violations_search, right_index=True, left_index=True)
# for index, row in df_reviews_search.iterrows():
#     if index in df_violations_search.index:
#         match_rows = df_violations_search.loc[:index]
df_merged.columns = ['yelp_name', 'hi_name']
df_merged.head()
        

yelp_name                        hi_name
1 s main st    subway            plaza hotel stuffed
1 s main st    subway               plaza steakhouse
1 s main st    subway         plaza steakhouse - bar
1 s main st    subway  plaza hotel veg prep / pantry
1 s main st    subway       plaza hotel derby lounge

In [313]:
df_merged.shape

(13773, 2)

In [314]:
from difflib import SequenceMatcher

def similar(a, b):
    return SequenceMatcher(None, a, b).ratio()

In [365]:
df_merged['sim_score']= df_merged.apply(lambda row: similar(a = row['yelp_name'], b = row['hi_name']), axis = 1)

In [367]:
df_reviews_search.loc['1 s main st']

name
address                             
1 s main st                   subway
1 s main st  center stage restaurant
1 s main st               mcdonald's
1 s main st             pop up pizza
1 s main st      zabas mexican grill
1 s main st              beer garden

In [371]:
df_reviews_search.loc['1 s main st'].shape

(6, 1)

In [373]:
df_violations_search.loc['1 s main st']

Restaurant Name
Address                                                       
1 s main st                                plaza hotel stuffed
1 s main st                                   plaza steakhouse
1 s main st                             plaza steakhouse - bar
1 s main st                      plaza hotel veg prep / pantry
1 s main st                           plaza hotel derby lounge
1 s main st                        plaza hotel in room service
1 s main st          hash house a go go - main kitchen/storage
1 s main st                           plaza hotel main kitchen
1 s main st                  island sushi & hawaiian grill bar
1 s main st                                             gigi's
1 s main st         island sushi & hawaiian grill sushi pantry
1 s main st                      island sushi & hawaiian grill
1 s main st           island sushi & hawaiian grill sushi prep
1 s main st  island sushi & hawaiian grill sushi hot produc...
1 s main st                         plaza hotel lounge @ plaza
1 s main st              plaza hotel simpatico service station
1 s main st                      plaza hotel 3rd floor kitchen
1 s main st                               plaza hotel dishroom
1 s main st                               plaza hotel pool bar
1 s main st                           plaza hotel showroom bar
1 s main st                        plaza hotel portable bar #1
1 s main st                           plaza hotel hot dog cart
1 s main st                          subway #22016 plaza hotel
1 s main st               plaza hotel bingo self serve station
1 s main st                               zaba's mexican grill
1 s main st                             mcdonald's #3153 plaza
1 s main st                        plaza hotel portable bar #2
1 s main st                        plaza hotel portable bar #3
1 s main st                        plaza hotel portable bar #4
1 s main st                        plaza hotel oscars main bar
1 s main st           plaza hotel oscars / simpatico food prep
1 s main st     plaza hotel oscars / simpatico service station
1 s main st        plaza hotel oscars / simpatico main kitchen
1 s main st                  hash house a go go - prep kitchen
1 s main st                    hash house a go go - restaurant
1 s main st                           hash house a go go - bar
1 s main st                      hash house a go go - hot line
1 s main st                   plaza hotel employee dining room
1 s main st                           plaza hotel omaha lounge
1 s main st                  plaza hotel race & sportsbook bar
1 s main st                                   biergarten - bar
1 s main st                                       pop up pizza
1 s main st                            biergarten - restaurant
1 s main st                        plaza hotel brightside cafe
1 s main st             plaza hotel brightside cafe - cookline

In [374]:
df_violations_search.loc['1 s main st'].shape

(45, 1)

In [370]:
df_merged.loc['1 s main st']

yelp_name                                            hi_name  \
1 s main st       subway                                plaza hotel stuffed   
1 s main st       subway                                   plaza steakhouse   
1 s main st       subway                             plaza steakhouse - bar   
1 s main st       subway                      plaza hotel veg prep / pantry   
1 s main st       subway                           plaza hotel derby lounge   
1 s main st       subway                        plaza hotel in room service   
1 s main st       subway          hash house a go go - main kitchen/storage   
1 s main st       subway                           plaza hotel main kitchen   
1 s main st       subway                  island sushi & hawaiian grill bar   
1 s main st       subway                                             gigi's   
1 s main st       subway         island sushi & hawaiian grill sushi pantry   
1 s main st       subway                      island sushi & hawaiian grill   
1 s main st       subway           island sushi & hawaiian grill sushi prep   
1 s main st       subway  island sushi & hawaiian grill sushi hot produc...   
1 s main st       subway                         plaza hotel lounge @ plaza   
1 s main st       subway              plaza hotel simpatico service station   
1 s main st       subway                      plaza hotel 3rd floor kitchen   
1 s main st       subway                               plaza hotel dishroom   
1 s main st       subway                               plaza hotel pool bar   
1 s main st       subway                           plaza hotel showroom bar   
1 s main st       subway                        plaza hotel portable bar #1   
1 s main st       subway                           plaza hotel hot dog cart   
1 s main st       subway                          subway #22016 plaza hotel   
1 s main st       subway               plaza hotel bingo self serve station   
1 s main st       subway                               zaba's mexican grill   
1 s main st       subway                             mcdonald's #3153 plaza   
1 s main st       subway                        plaza hotel portable bar #2   
1 s main st       subway                        plaza hotel portable bar #3   
1 s main st       subway                        plaza hotel portable bar #4   
1 s main st       subway                        plaza hotel oscars main bar   
...                  ...                                                ...   
1 s main st  beer garden              plaza hotel simpatico service station   
1 s main st  beer garden                      plaza hotel 3rd floor kitchen   
1 s main st  beer garden                               plaza hotel dishroom   
1 s main st  beer garden                               plaza hotel pool bar   
1 s main st  beer garden                           plaza hotel showroom bar   
1 s main st  beer garden                        plaza hotel portable bar #1   
1 s main st  beer garden                           plaza hotel hot dog cart   
1 s main st  beer garden                          subway #22016 plaza hotel   
1 s main st  beer garden               plaza hotel bingo self serve station   
1 s main st  beer garden                               zaba's mexican grill   
1 s main st  beer garden                             mcdonald's #3153 plaza   
1 s main st  beer garden                        plaza hotel portable bar #2   
1 s main st  beer garden                        plaza hotel portable bar #3   
1 s main st  beer garden                        plaza hotel portable bar #4   
1 s main st  beer garden                        plaza hotel oscars main bar   
1 s main st  beer garden           plaza hotel oscars / simpatico food prep   
1 s main st  beer garden     plaza hotel oscars / simpatico service station   
1 s main st  beer garden        plaza hotel oscars / simpatico main kitchen   
1 s main st  beer garden                  hash house a go go - pr

In [321]:
df_merged_sorted = df_merged.sort(columns = 'sim_score', axis = 0, ascending=False)
df_merged_sorted.head(n=200)

/Users/takumihouse/anaconda3/lib/python3.5/site-packages/ipykernel/__main__.py:1: FutureWarning: sort(columns=....) is deprecated, use sort_values(by=.....)
  if __name__ == '__main__':


yelp_name  \
2040 n nellis blvd                            arandas taqueria   
3981 e sunset rd                     capriotti's sandwich shop   
2893 w lake mead blvd                       mega tom's burgers   
2885 e charleston blvd                            pizza patron   
3945 s durango dr                   naked fish's sushi & grill   
3220 n jones blvd                               soul foo young   
3950 n tenaya way                           manhattan pizza iv   
3950 schiff dr                                   hunan jiu jia   
2850 e tropicana ave                             casa di amore   
2840 e tropicana ave                          bentley's coffee   
3989 spring mountain rd                       lee's sandwiches   
3920 spring mountain rd                               tofu hut   
4001 s decatur blvd                        shalimar restaurant   
4001 w sahara ave                                    los tacos   
4012 s rainbow blvd                              michael's pub   
4023 s maryland pkwy                                cafe verdi   
7315 w warm springs rd                  braddah's island style   
7315 arroyo crossing pkwy                          mimi's cafe   
7305 arroyo crossing pkwy                         jason's deli   
4808 s maryland pkwy                                 yoshinoya   
7435 s eastern ave                          ohana hawaiian bbq   
7480 s rainbow blvd                      cheeburger cheeburger   
780 e pyle ave                                   lefty's pizza   
5201 w charleston blvd                               los tacos   
7777 w sahara ave          four sevens sports bar & restaurant   
520 fremont st                              cheffinis hot dogs   
7770 w ann rd                       north beach italian bistro   
4830 blue diamond rd                         beanz coffee cafe   
7757 n el capitan way                                 del taco   
7720 s jones blvd                             mandarin express   
...                                                        ...   
8465 w sahara ave                          seattle dan's pizza   
8455 w lake mead blvd                           red rock pizza   
2590 s maryland pkwy                                china star   
4215 spring mountain rd                 sam woo bbq restaurant   
1105 e tropicana ave                            top this pizza   
4341 n rancho dr                             basil thai fusion   
1900 e charleston blvd                           tortas el rey   
1901 n decatur blvd                         mariscos el puerto   
1905 n decatur blvd                           five star tavern   
2238 n pecos rd                          la playita restaurant   
2237 n rampart blvd                        german bread bakery   
5300 spring mountain rd                          cafe de japon   
2227 n rampart blvd                     omelet house summerlin   
1910 village center cir                          samurai sam's   
6091 n decatur blvd                                burger king   
604 n rainbow blvd                           siam thai kitchen   
1930 fremont st                                  odyssey pizza   
3318 e flamingo rd                        zaba's mexican grill   
450 s buffalo dr                         southern style eatery   
4355 spring mountain rd                            tea station   
3555 s town center dr                                due forni   
4371 n rancho dr                               quickee burgers   
4371 n rancho dr                                southern style   
4375 n las vegas blvd                         little hong kong   
2267 n rampart blvd                         aloha island grill   
3331 e tropicana ave              chapala's mexican restaurant   
1801 e tropicana ave                               badger cafe   
2275 e sunset rd                                 pizza company   
1625 e charleston blvd                        church's chicken   
1675 industrial rd         

In [346]:
df_sim_threshold = df_merged_sorted[df_merged_sorted['sim_score'] > 0.69]
# df[df.a > 1]
df_sim_threshold.sort(columns = 'sim_score', axis = 0, ascending=True).head(n=20)

/Users/takumihouse/anaconda3/lib/python3.5/site-packages/ipykernel/__main__.py:3: FutureWarning: sort(columns=....) is deprecated, use sort_values(by=.....)
  app.launch_new_instance()


yelp_name  \
313 n nellis blvd       hibachi grill & supreme buffet   
5045 w tropicana ave               qdoba mexican grill   
4945 w flamingo rd                          mcdonald's   
7530 w washington ave                       mcdonald's   
4855 s maryland pkwy                        mcdonald's   
2233 n rampart blvd                          pizza hut   
7000 w charleston blvd                       pizza hut   
5111 boulder hwy                            mcdonald's   
8090 blue diamond rd                         pizza hut   
255 e flamingo rd                      tuscany gardens   
4505 w charleston blvd                      mcdonald's   
1195 e sahara ave                           mcdonald's   
5811 e charleston blvd                      mcdonald's   
4870 boulder hwy                             pizza hut   
6510 boulder hwy                             pizza hut   
3460 s hualapai way              the lodge at hualapai   
8820 w charleston blvd               three angry wives   
6505 n decatur blvd                chili's grill & bar   
9002 w sahara ave                             cafe rio   
3930 s las vegas blvd         1923 bourbon & burlesque   

                                                     hi_name  sim_score  
313 n nellis blvd                  hibachi grill - sushi bar   0.690909  
5045 w tropicana ave    qdoba mexican grill tropicana / deca   0.690909  
4945 w flamingo rd                          mcdonalds #17022   0.692308  
7530 w washington ave                       mcdonalds #25295   0.692308  
4855 s maryland pkwy                        mcdonalds #21201   0.692308  
2233 n rampart blvd                        pizza hut #026610   0.692308  
7000 w charleston blvd                     pizza hut #026588   0.692308  
5111 boulder hwy                            mcdonalds #23749   0.692308  
8090 blue diamond rd                       pizza hut #026596   0.692308  
255 e flamingo rd                                tuscany edr   0.692308  
4505 w charleston blvd                      mcdonalds #30210   0.692308  
1195 e sahara ave                           mcdonalds #23396   0.692308  
5811 e charleston blvd                      mcdonalds #23038   0.692308  
4870 boulder hwy                           pizza hut #026585   0.692308  
6510 boulder hwy                           pizza hut #026599   0.692308  
3460 s hualapai way             lodge at hualapai restaurant   0.693878  
8820 w charleston blvd      three angry wives pub restaurant   0.693878  
6505 n decatur blvd           chili's grill #1264 restaurant   0.693878  
9002 w sahara ave                            cafe rio sahara   0.695652  
3930 s las vegas blvd                 1923 bourbon stage bar   0.695652

In [347]:
df_sim_threshold.shape

(1233, 3)

In [117]:
df_lv_violations['Address'].map(len).max()

24

In [68]:
df_lv_violations.shape

(13890, 15)

In [41]:
for x in business_locations:
    print(x)

C3WAtjq006LVC-EHCt-fRA
4O9Du_JiAvz24b2VSOJNTA
jWiGtY7hTqBjbzN3yyRXbw
IV-1NJnGA3O0ZRBrZEWWaA
V4XYRuhq_614tFFvWK5q6g
jgkT9RA3OCBqgniWCkwLrA
-GF267VCCEZ7es8pnKxRoA
1nO6ffHEwZYFBzEsyw1w7w
FV16IeXJp2W6pnghTz2FAw
0BVxe-S5aZMa0GgIYqDmHQ
8zTes4kUSoM161AD7G6DsQ
w29d_3Ua0_JlOaAk89Sr2w
O9ro4MvZ7LDQC76zLEg7AQ
UQEf2T_XYszW2F_y7rBNLA
fuCKyA_ZTXVEo8BwUaeh1Q
tl-yAtKfAB7wtCWehVH1cQ
aGbjLWzcrnEx2ZmMCFm3EA
mb4B1LdcEkz4V6BCm-LeFw
HXmKSaJa7hXsgeCozQs-uQ
XpdKTCAPOSUj50fTnW94aA
OK-v9HfG53SCFvZ4mn5IYQ
1EF9J1TGjxX5gXB3U8hjvA
gM8YcvjG2G8CTgCarkBAeA
uXJbNkU6cKAooMWTWnl5aA
3Igoz7noWksymLqoMWuzRg
shA3pC1JuAkXl28_LVcp7w
O8BRzdXLBzu6K4Sm32fxng
6ZaA_KOS1wW5PqDPAHH9Qg
FDfK4HygC_P-Idoy_K8PKQ
-CIZh4JOZuOEbSzLVrAIGw
YO-hq425916TWm4Rj2-Y-A
N0sUdCBcA3ScYULAkGOqxA
pAWgUsCkPwJG4Vw75Wk0aw
w8sRfLP8U7uX4OktcSbb_w
CfdCsVkzzM3uJKGRzA0ufA
957AgCsV5hhFLDJ5aw546w
cWFpyN49FRsFrNd3_cb50g
XQvKQDAyE3vLsDdjJBKs4Q
X8P2i2tyhLSzljdliBFBrg
sAbRiVqmxNdjTja_vmnsnA
oy90Dd3UjsyAmj_lxSda5g
8buIr1zBCO7OEcAQSZko7w
TNIR0ht13OSqG-HlkoG6uQ
gPmoQaWDiVn

In [59]:
# create lists for addresses of businesses with violations in order to 
# check for overlap between the Las Vegas Violation set and the yelp set.

violation_addresses = []

for line in df_agg['Address']:
    violation_addresses.append({df_agg[str(line).lower())
violation_addresses[:10]

#overlap_df_revs=df_agg_revs[df_agg_revs['address'].isin(violation_addresses)]
#overlap_df_agg=df_agg[df_agg['Address'].isin(overlap_df_revs['address'])]
#overlap_df_revs=overlap_df_revs.reset_index(drop = True)
#overlap_df_revs

SyntaxError: invalid syntax (<ipython-input-59-e0eeecb2a60b>, line 7)

In [59]:
overlap_df_agg

Empty DataFrame
Columns: [Restaurant Name, Permit Number, Location Name, Category Name, Address, Current Demerits, Current Grade, Date Current, Inspection Demerits, Inspection Grade, Permit Status, Inspection Result, Violations, X, Y]
Index: []

In [31]:
def compare()

911


It looks like there are a total of 911 unique restaurant addresses that have an evaluated sick score greater than 0 and entries in the inspection dataset.

In [32]:
categories = df_agg.columns
df_agg = df_agg.drop(['Inspection Grade'], axis=1)
df_agg = df_agg.drop(['Inspection Demerits'], axis=1)              

In [33]:
df_agg

Restaurant Name Permit Number  \
0                                   #1 HAWAIIAN BARBEQUE     PR0110047   
1                                                $5 CAFE     PR0017837   
2                    100 DEGREES HOT POT - DELETED 10/15     PR0113718   
3                                     101 MART SNACK BAR     PR0118514   
4                                    168 Market - Bakery     PR0021447   
5                                168 Market - Restaurant     PR0021446   
6                        168 Market / Produce Department     PR0021445   
7                             17 SOUTH BOOZE & BITES BAR     PR0109450   
8                           17 SOUTH BOOZE & BITES GRILL     PR0109449   
9                                     1900 ASIAN CUISINE     PR0099062   
10                                      1923 BOURBON BAR     PR0110263   
11                       1923 BOURBON ISLAND BAR/SUPPORT     PR0108949   
12                                1923 BOURBON STAGE BAR     PR0108950   
13                                  1ST FLOOR BAR- BOMAN     PR0021200   
14                          1ST FLOOR EAST BAR- REYNOLDS     PR0021207   
15                            1ST FLOOR PANTRY- REYNOLDS     PR0021208   
16                          1ST FLOOR WEST BAR- REYNOLDS     PR0021206   
17                                 2 AMIGOS TACOS MOBILE     PR0119469   
18                                            215 Saloon     PR0024098   
19                           24 HOUR MINI MART SNACK BAR     PR0106164   
20                          24 Hour Fitness Smoothie Bar     PR0012767   
21                                             24 LOUNGE     PR0098970   
22                              24X7 MINI MART SNACK BAR     PR0021102   
23                                   28Go the Restaurant     PR0024320   
24                                  2ND FLOOR BAR- BOMAN     PR0021201   
25                      2ND FLOOR FOUNDERS BAR- REYNOLDS     PR0021209   
26                    2ND FLOOR FOUNDERS PANTRY REYNOLDS     PR0021210   
27                     2ND FLOOR MEZZ GRAB N GO-REYNOLDS     PR0104678   
28                    2ND FLOOR MEZZ NORTH BAR- REYNOLDS     PR0021212   
29                       2ND FLOOR MEZZ PANTRY- REYNOLDS     PR0021213   
...                                                  ...           ...   
13860                               ZEFFIRINO RESTAURANT     PR0009057   
13861                            ZEFFIRINO SERVICE BAR 1     PR0009059   
13862                         ZENSHIN ASIAN PREP KITCHEN     PR0105401   
13863                           ZENSHIN ASIAN RESTAURANT     PR0098470   
13864                            ZENSHIN ASIAN SUSHI BAR     PR0098471   
13865     ZENSHIN ASIAN WAREWASH/BEV STATION/DRY STORAGE     PR0105403   
13866                            ZEST BISTRO & BAR - BAR     PR0017887   
13867                   ZEST BISTRO & BAR - PREP KITCHEN     PR0098325   
13868                     ZEST BISTRO & BAR - RESTAURANT     PR0017886   
13869                                 ZIKIZ COFFEE HOUSE     PR0120264   
13870                      ZINGERS BEST BURGERS ON EARTH     PR0014610   
13871                  ZINGERS BEST BURGERS ON EARTH BAR     PR0014611   
13872                             ZOMBIE DOGS @ HARD HAT     PR0118834   
13873  ZOMBIE DOGS PUSF - DELETED 7/29/16 APPLYING FO...     PR0109033   
13874                                          ZONA ROSA     PR0020519   
13875                                     ZYDECO PO-BOYS     PR0112724   
13876                 Zaytoon Mediterranean - Restaurant     PR0020288   
13877                                          Zizzy Bar     PR0015979   
13878                                   Zizzy Restaurant     PR0015978   
13879                                      Zodies Lounge     PR0007654   
13880              Zoological - Botanical Park Snack Bar     PR0007240   
13881                 at Your Service Catering Snack Bar     PR0005647   
13882                               in N Out 

In [73]:
df_cat = pd.get_dummies(df_agg['Category Name'], prefix = 'category')
df_grade = pd.get_dummies(df_agg['Current Grade'], prefix = 'grade_cur')
df_perm = pd.get_dummies(df_agg['Permit Status'], prefix = 'permit')
df_res = pd.get_dummies(df_agg['Inspection Result'], prefix = 'result')
df_dems = pd.get_dummies(df_agg['Current Demerits'], prefix = 'demerits')
df_type = pd.get_dummies(df_agg['Inspection Type'], prefix = 'type')
df_new = pd.concat([df_agg, df_cat], axis=1)
df_new = pd.concat([df_new, df_grade], axis=1)
df_new = pd.concat([df_new, df_perm], axis=1)
df_new = pd.concat([df_new, df_res], axis=1)
df_new = pd.concat([df_new, df_dems], axis=1)
df_new = pd.concat([df_new, df_type], axis=1)
df_new = df_new.drop(['Category Name'], axis = 1 )
df_new = df_new.drop(['Current Grade'], axis = 1 )
df_new = df_new.drop(['Permit Status'], axis = 1 )
df_new = df_new.drop(['Inspection Result'], axis = 1)
df_new = df_new.drop(['Current Demerits'], axis = 1)
df_new = df_new.drop(['Inspection Type'], axis = 1)
df_new = df_new.drop(['Date Current'], axis = 1)
df_new = df_new.drop(['Restaurant Name'], axis = 1)
df_new = df_new.drop(['Permit Number'], axis = 1)
df_new = df_new.drop(['Location Name'], axis = 1)
df_new = df_new.drop(['Address'], axis = 1)
df_new.fillna(0)

KeyError: 'Inspection Type'

In [ ]:
df_new

In [60]:
from sklearn.neighbors import KNeighborsClassifier
from sklearn.cross_validation import train_test_split

X = df_new.loc[:,:].values
y = df_new['grade_cur_X'].values
X_train, X_test, y_train, y_test = train_test_split(X, y, )
knn = KNeighborsClassifier(n_neighbors=4)
knn.fit(X_train,y_train)
pred = knn.predict(X_test)
print(pred)
test_score = knn.score(X_test,y_test)
print('Accuracy on test data: {}'.format(test_score))


[ 0.  0.  0. ...,  0.  0.  0.]
Accuracy on test data: 0.9997120644975526


In [62]:
X = df_new.loc[:,:].values
y = df_new['grade_cur_A'].values
X_train, X_test, y_train, y_test = train_test_split(X, y)
knn = KNeighborsClassifier(n_neighbors=4)
knn.fit(X_train,y_train)
pred = knn.predict(X_test)
print(pred)
test_score = knn.score(X_test,y_test)
print('Accuracy on test data: {}'.format(test_score))

[ 1.  1.  1. ...,  1.  1.  1.]
Accuracy on test data: 0.9979844514828679
